scripts/inspect_ckpt.py

```
@@ -0,0 +1,229 @@
+    H("6. TimestepEmbedding weights")
+    te_keys = sorted(k for k in meta if "time_embed" in k)
+    for k in te_keys:
+        s = load_and_measure(bp, k)
+        W(f"  {k}: shape={s['shape']}, mean={s['mean']:.6f}, std={s['std']:.6f}")
+
+    # ---- 7. Per-layer param counts (metadata only) ----
+    H("7. DECODER per-layer param counts (metadata, zero RAM)")
+    for li in range(24):
+        pf = f"decoder.layers.{li}."
+        lk = {k: v for k, v in meta.items() if k.startswith(pf)}
+        nparams = sum(eval("*".join(str(s) for s in v.get("shape",[])) or "0") for v in lk.values())
```

Ruff (0.15.7)

[error] Use of possibly insecure function; consider using `ast.literal_eval`

(S307)